src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -1141,6 +1141,19 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
                 args.reserve(al, n_pos_args + n_kwargs);
                 visit_expr_list(pos_args, n_pos_args, kwargs, n_kwargs,
                                 args, rt_subs, func, loc);
+            }
+        	if((n_pos_args+ n_kwargs) < func->n_args && (args.size() < func->n_args) ){
```

Comment:
```suggestion
        if ((n_pos_args + n_kwargs) < func->n_args && (args.size() < func->n_args) ){
```

Reply:
Also, I think `n_pos_args + n_kwargs` and `args.size()` will be same. So checking for `args.size() < func->n_args` seems redundant.

Reply:
I wrote this part in the first time as you said, but built-in function  `complex()` made a huge problem.
The function seems to be making a call to some function without passing any positional arguments to the call ,although the call has `Vec<ASR::call_arg_t>`  parameter filled with arguments to be placed in the called function.
# Brief 
- Sometimes the call is depending on `n_pos_args + n_kwargs` as you mentioned and that should be enough, but sometimes it depends on `args.size()` .
- `n_pos_args + n_kwargs` evaluate to => 0 .... while 'args.size()' evaluate to => (**larger than 0**).

Reply:
I see. I think in that case, we can directly use `if (args.size() < func->n_args)`. Can you try it and share if it works?

You also need to update the initialization in the below loop `for(size_t def_arg = args.size(); def_arg < func->n_args; def_arg++)`

Reply:
Yeah, you're right. Using `arg.size()` only would do the job.